### PR TITLE
#150398520 Enable user to export and import workouts

### PR DESCRIPTION
**Pull request by jmnyarega**
**What this PR does?**
- Allows users to send their workouts to other users so they can add them in their workout schedules.
- Allows transfer of workouts between instances

 **What are the relevant tickets?**
- #150398520

### CODE REVIEW

File: wger/manager/templates/workout/overview.html [hunk 1]
```diff
@@ -1,14 +1,7 @@
...
-
-
-{% block content %}
-    <div class="list-group">
-        {% for workout in workouts %}
-            <a href="{{ workout.get_absolute_url }}" class="list-group-item">
+{% extends "base.html" %} {% load i18n %} {% load staticfiles %} {% block title %}{% trans "Your workouts" %}{% endblock %} {% block content %}
+<div class="list-group">
+    {% for workout in workouts %}
+    <a href="{{ workout.get_absolute_url }}" class="list-group-item">
```
Comment (egichuri) [inline wger/manager/templates/workout/overview.html]: any reason the styling was changed here?

File: wger/manager/templates/workout/view.html [hunk 2]
```diff
@@ -119,30 +119,30 @@
...
-            </div>
-            <div class="modal-footer">
-              <button type="button" class="btn btn-default" data-dismiss="modal">{% trans "Close" %}</button>
-            </div>
-        </div>
+<div class="modal-dialog">
+  <div class="modal-content">
+    <div class="modal-header">
+      <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+      <h4 class="modal-title">{% trans "Export calendar file" %}</h4>
```
Comment (egichuri) [inline wger/manager/templates/workout/view.html:126]: What's the change here?

File: wger/manager/views/workout.py [hunk 3]
```diff
@@ -53,6 +57,44 @@
...
+        # send response to data.json file
+        response['Content-Disposition'] = 'attachment; filename=data.json'
+        return response
+
+
+def import_workouts(request):
+    file = request.FILES['import_file']
+    data = file.read()
+    f = open('data.json', 'r')
+    import json
```
Comment (egichuri) [inline wger/manager/views/workout.py]: this has already been imported

File: wger/manager/views/workout.py [hunk 4]
```diff
@@ -53,6 +57,44 @@
...
+        response = HttpResponse(data, content_type='text/json')
+        # send response to data.json file
+        response['Content-Disposition'] = 'attachment; filename=data.json'
+        return response
+
+
+def import_workouts(request):
+    file = request.FILES['import_file']
+    data = file.read()
+    f = open('data.json', 'r')
```
Comment (egichuri) [inline wger/manager/views/workout.py]: seems like you're reading from a hard-coded file here. is the `data` assigned on [line 78](https://github.com/andela/wger-lycan-clan/pull/10/files#diff-2923ec76a8e1150a8800e979bdee2de0R78) used anywhere?

File: wger/manager/views/workout.py [hunk 5]
```diff
@@ -53,6 +57,44 @@
...
+    json1_data = json.loads(f.read())
+    for value in json1_data:
+        value['fields']['user'] = request.user.id
+        value['pk'] = random.randint(50, 1909)
+    for deserialized_object in serializers.deserialize("json", json.dumps(json1_data)):
+        deserialized_object.save()
+
+    template_data = {}
+
+    workouts = Workout.objects.filter(user=request.user)
```
Comment (egichuri) [inline wger/manager/views/workout.py]: This is already done in `overview`. No need to repeat it here. [3 Key Software Principles You Must Understand](https://code.tutsplus.com/tutorials/3-key-software-principles-you-must-understand--net-25161)

File: wger/manager/views/workout.py (hunk 4 above)
Comment (egichuri) [reply wger/manager/views/workout.py]: Also check usage of `with` when dealing with files

File: wger/manager/views/workout.py [hunk 6]
```diff
@@ -53,6 +57,44 @@
...
+def import_workouts(request):
+    file = request.FILES['import_file']
+    data = file.read()
+    f = open('data.json', 'r')
+    import json
+    json1_data = json.loads(f.read())
+    for value in json1_data:
+        value['fields']['user'] = request.user.id
+        value['pk'] = random.randint(50, 1909)
+    for deserialized_object in serializers.deserialize("json", json.dumps(json1_data)):
```
Comment (egichuri) [inline wger/manager/views/workout.py]: Ideally you should be adding/creating the imported workouts. Not really sure what's going on here. Please check the `WorkoutEditView` class and `add`/`copy_workout` methods on this file to get an idea of how you should do this.

**Review by egichuri — CHANGES_REQUESTED** [review]

**Comment by egichuri** [comment]
Any screenshots to show these changes?

File: wger/manager/views/workout.py (hunk 6 above)
Comment (jmnyarega) [reply wger/manager/views/workout.py]: The script above is adding imported workouts, The workouts that were exported from  another instance

File: wger/manager/views/workout.py (hunk 3 above)
Comment (jmnyarega) [reply wger/manager/views/workout.py]: Please elaborate on this, Imported from where?

**Review by egichuri — APPROVED** [review]

**Pull request merged**